### ISSUE TITLE
Defects reported by scan-build in macOS

### ISSUE THREAD

**Issue by Skeptor** [wazuh/wazuh#3774]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9 | Syscollector | Agent | Packages/Sources | MacOS |

The compilation of the code with the tool `scan-build` in a MacOS agent has reported two memory leaks in Syscollector:

`Memory Error | Memory leak | syscollector_bsd.c:572`:
This one is related to the variable: `serial_str`
`Memory Error | Memory leak | syscollector_bsd.c:586`:
This one is related to the variable: `serial`

### Update

Branch 3.9 doesn't contain the errors reported by scan-build [here](https://github.com/wazuh/wazuh/issues/3450) and fixed at the PR #3450 merged in 3.10.

`scan-build` in macOS for the 3.9 branch reports 66 bugs. However, in 3.10 it reports 12 bugs that we have to analyze and solve. Here the report:

| tool | version |
|---|---|
| macOS | 10.14 mojave |
| scan-build | checker-279 |
| clang version | 4.0.0 |
| make version | 3.81 |

**Report**:

```
scan-build: 12 bugs found.
scan-build: Run 'scan-view /tmp/scan-build-2019-07-29-192630-10820-1' to examine bug reports.
```

Bug Group | Bug Type | File | Function/Method | Line |
-- | -- | -- | -- | -- 
Unix API | Allocator sizeof operand mismatch | wazuh_modules/syscollector/syscollector_bsd.c | getGatewayList | 1255
API | Argument with 'nonnull' attribute passed null | os_execd/config.c | getClusterConfig | 331
Dead store | Dead assignment | addagent/main.c | main | 175
Logic error | Dereference of null pointer | os_xml/os_xml.c | _xml_sgetc | 52
Logic error | Dereference of null pointer | os_regex/os_regex_execute.c | OSRegex_Execute_ex | 118
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | sys_hw_bsd | 586
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | getGatewayList | 1307
Memory Error | Memory leak | syscheckd/create_db.c | read_dir | 760
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | sys_hw_bsd | 572
Memory error | Nullability | wazuh_modules/wm_osquery_monitor.c | wm_osquery_monitor_main | 619
Unix API | Undefined allocation of 0 bytes (CERT MEM04-C; CWE-131) | wazuh_modules/wm_control.c | getPrimaryIP | 51
Logic error | Uninitialized argument value | wazuh_modules/wm_exec.c | wm_exec | 298

From the list above, fails related to `getGatewayList()` and `getPrimary()` are solved in #3773.

**Comment by JuantAldea** [comment]
- Logic error	Uninitialized argument value	wazuh_modules/wm_exec.c	wm_exec	298	
   *  Because boolean expresions use Short-circuit evaluation, if output evaluates to false pipe() is not executed and w_descriptor_cloexec(pipe_fd[0]); is called with an undefined pipe_fd.

**Comment by JuantAldea** [comment]
Bug Group | Bug Type | File | Function/Method | Line | Fixed in
-- | -- | -- | -- | -- | --
Unix API | Allocator sizeof operand mismatch | wazuh_modules/syscollector/syscollector_bsd.c | getGatewayList | 1255 | #3773
API | Argument with 'nonnull' attribute passed null | os_execd/config.c | getClusterConfig | 331 | COMMIT_ID
Dead store | Dead assignment | addagent/main.c | main | 175 | COMMIT_ID
Logic error | Dereference of null pointer | os_xml/os_xml.c | _xml_sgetc | 52 | Won't fix. Also disappears when scanning with a newer llvm
Logic error | Dereference of null pointer | os_regex/os_regex_execute.c | OSRegex_Execute_ex | 118 | Won't fix. Also disappears when scanning with a newer llvm
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | sys_hw_bsd | 586 | COMMIT_ID
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | getGatewayList | 1307 | #3773
Memory Error | Memory leak | syscheckd/create_db.c | read_dir | 760 | COMMIT_ID
Memory Error | Memory leak | wazuh_modules/syscollector/syscollector_bsd.c | sys_hw_bsd | 572 | COMMIT_ID
Memory error | Nullability | wazuh_modules/wm_osquery_monitor.c | wm_osquery_monitor_main | 619 | COMMIT_ID
Unix API | Undefined allocation of 0 bytes (CERT MEM04-C; CWE-131) | wazuh_modules/wm_control.c | getPrimaryIP | 51 | #3773
Logic error | Uninitialized argument value | wazuh_modules/wm_exec.c | wm_exec | 298 | COMMIT_ID

**Comment by chemamartinez** [comment]
Merged #3785
